modules/Planner/planner_view_full.php

```diff
@@ -1172,7 +1172,25 @@
 
                         // Display the date this attendance was taken, if any
                         if ($canTakeAttendance) {
-                            $offTimetableStudents = $container->get(SchoolYearSpecialDayGateway::class)->selectOffTimetableStudentsByClass($session->get('gibbonSchoolYearID'), $gibbonCourseClassID, $values['date'])->fetchKeyPair();
+                            // BEGIN: Guard against a crash for the missing method
+							// Before (crashes if method doesn't exist):
+							// $offTimetableStudents = $container->get(\Gibbon\Domain\School\SchoolYearSpecialDayGateway::class)
+							//     ->selectOffTimetableStudentsByClass($session->get('gibbonSchoolYearID'), $gibbonCourseClassID, $values['date'])
+							//     ->fetchKeyPair();
+
+							$offTimetableStudents = [];
+							$specialDayGateway = $container->get(\Gibbon\Domain\School\SchoolYearSpecialDayGateway::class);
+
+							if (method_exists($specialDayGateway, 'selectOffTimetableStudentsByClass')) {
```

Comment:
Hmm. If your Gibbon install is fully up to date, this method_exists check shouldn't be necessary, as the method does indeed exist. The SchoolYearSpecialDayGateway is included with a `use` statement at the top of the file, so it shouldn't need fully qualified.